### PR TITLE
[INLONG-8266][Manager] Optimize PostgreSQL field type mapping strategy with the customized configuration file

### LINKED ISSUE #8266: [Improve][Manager] Optimize PostgreSQL field type mapping strategy with the customized configuration file
### Description

Optimize PostgreSQL field type mapping strategy by replacing enum class to the customized configuration file

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by chestnut-c**
[INLONG-8266][Manager] Optimize PostgreSQL field type mapping strategy with the customized configuration file

- Fixes #8266

### Motivation

Optimize PostgreSQL field type mapping strategy by replacing enum class to the customized configuration file,easy to expand it.

### CODE REVIEW

File: inlong-manager/manager-common/pom.xml [hunk 1]
```diff
@@ -123,6 +123,10 @@
             <groupId>org.apache.poi</groupId>
             <artifactId>poi-ooxml</artifactId>
         </dependency>
+        <!-- parse yaml -->
+        <dependency>
+            <groupId>com.esotericsoftware.yamlbeans</groupId>
+            <artifactId>yamlbeans</artifactId>
```
[the remark this answers is not in the document]
Comment (chestnut-c) [reply inlong-manager/manager-common/pom.xml]: > Please add yamlbean's license file and notice file under /licenses/inlong-manager/ directory.

ok, i replace yamlbeans with snakeyaml maven dependency of apache v2 protocol and add to LICENSE.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @param typeMapping type mapping map
+     */
+    @SuppressWarnings("unchecked")
+    private void readerOption(Map<?, ?> converterConf, String key, Map<String, String> typeMapping) {
+        if (converterConf.containsKey(key)) {
+            List<Map<String, String>> typeMappings = (List<Map<String, String>>) converterConf.get(key);
+            for (Map<String, String> mapping : typeMappings) {
+                String sourceType = mapping.get("source.type");
+                String targetType = mapping.get("target.type");
+                if (StringUtils.isEmpty(sourceType) || StringUtils.isEmpty(targetType)) {
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java:90]: StringUtils.isBlank()

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     * @param converterConf converterConf
+     * @param key key in converter file
+     * @param typeMapping type mapping map
+     */
+    @SuppressWarnings("unchecked")
+    private void readerOption(Map<?, ?> converterConf, String key, Map<String, String> typeMapping) {
+        if (converterConf.containsKey(key)) {
+            List<Map<String, String>> typeMappings = (List<Map<String, String>>) converterConf.get(key);
+            for (Map<String, String> mapping : typeMappings) {
+                String sourceType = mapping.get("source.type");
```
Comment (vernedeng) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java:88]: It's better to make them constants

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java (hunk 3 above)
Comment (chestnut-c) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java]: > It's better to make them constants

okk

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**